NuRadioReco/modules/io/coreas/readCoREASShower.py

```diff
@@ -68,7 +70,12 @@ def run(self):
 
             f_coreas = corsika["CoREAS"]
 
-            evt = NuRadioReco.framework.event.Event(corsika['inputs'].attrs['RUNNR'], corsika['inputs'].attrs['EVTNR'])
+            if set_ascending_run_and_event_number:
+                evt = NuRadioReco.framework.event.Event(self.__ascending_run_and_event_number, self.__ascending_run_and_event_number)
+                self.__ascending_run_and_event_number += 1
+            else:
+                evt = NuRadioReco.framework.event.Event(corsika['inputs'].attrs['RUNNR'], corsika['inputs'].attrs['EVTNR'])
```

Comment:
shouldn't we be robust here and check if we already have an event with the same id? Actually this should be checked by the eventWriter. But I think we should catch this. This might be a really difficult bug to find in the future. So better do it now properly.